File: js/therapy-v3/src/types/specialty.ts
export interface Specialty {
  id: number;
  name: string;
  description: string | null;
  duration_min: number;
  color: string;
  active: boolean;
  max_slots: number;
  available_slots: number;
  schedule_days: number[] | null;
  schedule_start: string | null;
  schedule_end: string | null;
  created_at: string;
  updated_at: string;
}

export interface SpecialtyCreateInput {
  name: string;
  description?: string | null;
  duration_min?: number;
  color?: string;
  max_slots?: number;
  schedule_days?: number[] | null;
  schedule_start?: string | null;
  schedule_end?: string | null;
}

export interface SpecialtyUpdateInput {
  name?: string;
  description?: string | null;
  duration_min?: number;
  color?: string;
  max_slots?: number;
  schedule_days?: number[] | null;
  schedule_start?: string | null;
  schedule_end?: string | null;
}

File: python/therapy-api/src/therapy/specialty/domain/model/specialty.py
from dataclasses import dataclass
from datetime import datetime


@dataclass(frozen=True)
class Specialty:
    name: str
    id: int = 0
    slug: str = ""
    description: str | None = None
    duration_min: int = 60
    color: str = "#7B8C76"
    active: bool = True
    max_slots: int = 1
    available_slots: int = 1
    schedule_days: list[int] | None = None
    schedule_start: str | None = None
    schedule_end: str | None = None
    created_at: datetime | None = None
    updated_at: datetime | None = None
